src/ppl/src/IOPlacer.tcl

```diff
@@ -336,6 +336,25 @@ proc set_simulated_annealing { args } {
   ppl::set_simulated_annealing $temperature $max_iterations $perturb_per_iter $alpha
 }
 
+sta::define_cmd_args "simulated_annealing_debug" {
+  [-iters_between_paintings iters]
+  [-no_pause_mode no_pause_mode] # Print solver state every second based on iters_between_paintings
+}
+
+proc simulated_annealing_debug { args } {
+  sta::parse_key_args "simulated_annealing_debug" args \
+  keys {-iters_between_paintings} \
+  flags {-no_pause_mode}
+
+  set no_pause_mode [info exists flags(-no_pause_mode)]
+
+  if [info exists keys(-iters_between_paintings)] {
```

Comment:
This looks to be mandatory.  Is that intentional?  If so, an error should be given if not provided.  If not there should be a default value.

Reply:
It's intentional. I see the errors have numbers, (e. g. PPL 17). Which number must I use?

Reply:
Look at the file `ppl/messages.txt`. You must choose a number that is not present in this file yet.